templates/components/what/networking/post-json.hbs

```diff
@@ -1,10 +1,15 @@
 // This will POST a body of
-//     `{"lang":"rust","body":"json"}`
-let mut map = HashMap::new();
-map.insert("lang", "rust");
-map.insert("body", "json");
+//     `{"delete": true, "user": 42}`
+#[derive(Serialize)]
+struct Body {
+    delete: bool,
```

Comment:
I feel like keeping at least one string example in the struct could be useful. Strings are pretty ugly in most low-level languages and json deals with them a lot. Showing that Rust _doesn't_ have these issues could be good

Reply:
I agree in principle, but using a string would require
* Introducing lifetimes very early (when using a `&'a str` as a struct field), or
* Making use of `.into()`/`.to_string()` conversion to get a `String`

(using `String` was the previous idea for this, see https://github.com/rust-lang/www.rust-lang.org/issues/482)

I think I'll let @aturon decide whether we should change. I don't really have a strong opinion here other than "we should totally use serde".

Reply:
Hmm, yea. I see your points. Although considering that lifetimes are quite a central part of Rust, I'm not sure that we should hide their existence from code examples on the website *too much* either.